contracts/Market.sol

```diff
@@ -70,7 +74,13 @@ contract Market is IMarket, Ownable, Pausable {
 
     // The beneficiary can retrieve already released funds via this.
     function withdrawAvailableFunds() external payable {
-        _beneficiary.transfer(this.balance);
+        if (totalWithdrawn >= beneficiaryLimit)
+            owner().transfer(this.balance);
+        else {
+            uint256 amonut = Math.min(this.balance, beneficiaryLimit);
```

Comment:
I feel like this could be expressed more cleanly:

```solidity
uint256 amountToBeneficiary = Math.min(this.balance - totalWithdrawn, beneficiaryLimit)
uint256 amountToOwner = this.balance - amountToBeneficiary
totalWithdrawn += amountToBeneficiary
_beneficiary.transfer(amountToBeneficiary)
owner().transfer(amountToOwner)
```

Reply:
This can be called repeatedly. It is a lot of extra gas doing those calls with 0 eth.